libopae/plugins/xfpga/metrics/bmc/CMakeLists.txt

```diff
@@ -27,7 +27,7 @@
 include_directories(
 	${OPAE_INCLUDE_DIR} )
 
-add_library(bmc SHARED
+add_library(bmc MODULE
```

Comment:
Are you certain of this? libbmc.so is used on the link line, so it should be SHARED instead of MODULE.  MODULEs are loaded with dlopen and can't be specified on a link line (-lbmc).

Reply:
It is dlopen'ed on metrics/metrics_utils.c:857

Reply:
But it is (I believe) linked with pacd and fpgainfo (doesn't fpgad use this?)

Reply:
I looked at fpgainfo's CMakeLists.txt.  It isn't linking with bmc.  If pacd is specifying -lbmc, is there any way it can be loaded dynamically instead.  fpgad isn't using the metrics api atm.  If we make such updates in the future, I think it will access libbmc.so by way of libxfpga.so only.

Reply:
The bmc CMakeLists.txt has provisions for building the library as libstaticbmc.a.  That might be another option.

Reply:
I think at this point, we should make it a module and any applications that need it can link to bmcstatic.